### DISCUSSION TITLE
Allow to skip `prepare-commit-msg` hook

### DISCUSSION THREAD

**Discussion by foosinn** [magit/magit#4855] [Ideas]
Hey,

some of our devs use https://github.com/ramsey/conventional-commits.

This adds an interactive `prepare-commit-msg` hook if the message is empty. Sadly this is breaking `magit`.

Is it possible to skip the hook? I've browsed through docs and issues, but could not find anythink.

Thanks!

**Comment by twmr** [comment]
Have you tried using "-n" in the magit-commit popup? It should skip all pre-commit and commit-msg hooks.

**Reply by foosinn** [reply]
Yes, but this seems to not skip `pepare-commit-msg` hooks. Also it would be awesome to always skip this.

**Reply by twmr** [reply]
from: https://git-scm.com/docs/githooks
> The purpose of the hook is to edit the message file in place, and it is not suppressed by the --no-verify option

Yes, it seems as if it is not skipped. 🤔

**Reply by twmr** [reply]
I've just found https://stackoverflow.com/questions/42992459/can-i-suppress-git-hook-e-g-prepare-commit-msg-via-git-command-line-argument

**Reply by foosinn** [reply]
~~`SKIP=prepare-commit-msg` works for me. Would be awesome if one could set it in magit but setting in emacs environment is fine, too. :)~~

`SKIP` seems to not work at all o.O

```
❯ SKIP=prepare-commit-msg git commit
pre-commit:
 - hook is disabled
prepare-commit-msg:
 - \Ramsey\CaptainHook\PrepareConventionalCommit                     :
Prepare Commit Message
======================

 The following prompts will help you create a commit message that
 follows the Conventional Commits specification.

 What is the type of change you're committing? (e.g., feat, fix, etc.) [feat]:
 >
 ```

The custom git setting is a nice trick, but since the tools control the hook files this won't work.

Thanks for your time
